workspaces/spectacle/src/index.ts

```diff
@@ -311,6 +312,35 @@ export async function makeSpectacle(opticContext: IOpticContext) {
           .spectacleContext()
           .opticContext.diffRepository.findById(diffId);
       },
+      metadata: async (parent: any, args: any, context: any, info: any) => {
+        const metadataKey = 'metadata';
+        const specIdKey = 'id';
+        const metadata =
+          context.spectacleContext().contributionsProjection[metadataKey] || {};
+        let specId = metadata[specIdKey];
+
+        if (!specId) {
+          specId = uuidv4();
+          await context
+            .spectacleContext()
+            .opticContext.specRepository.applyCommands(
+              [
+                {
+                  AddContribution: {
+                    id: metadataKey,
+                    key: specIdKey,
+                    value: specId,
+                  },
+                },
+              ],
+              uuidv4(),
+              'Set spec metadata ID',
+              { clientId: '', clientSessionId: '' }
```

Comment:
Is there an (easy way) to add a contribution without creating a batch commit? Ideally this is transparent and should not show up in history? Or would this be an antipattern (especially since we are using batch commits as _any_ change to a spec)?

Reply:
I think it's fine to put it in a batch commit. but we can definitely expose a way that doesn't. We can put a link to docs on the metadata id in the commit message so it's not confusing / there's an external reference for what it is and why it exists. if we change contributions, I believe these will stay, and then we'd have ExternalContributionAdded(contentUrl or contentPath, contentChecksum) or something like that

Reply:
I think I'm going to change this to something like "initialize specification attributes" - on existing specs, this could be confusing. But for new specifications this would make sense